backend/cmm_helpers.ml

```diff
@@ -1161,6 +1161,26 @@ let unboxed_mutable_float_array_ref arr ofs dbg =
 let unboxed_immutable_float_array_ref arr ofs dbg =
   Cop (mk_load_immut Double, [array_indexing log2_size_float arr ofs dbg], dbg)
 
+let unboxed_mutable_float32_unboxed_product_array_ref arr ~element_num dbg =
+  bind "arr" arr (fun arr ->
+      bind "index" element_num (fun index ->
+          Cop
+            (* XXX mshinwell for mslater: is Float32 correct? *)
+            ( mk_load_mut (Single { reg = Float32 }),
+              [array_indexing log2_size_float arr index dbg],
```

Comment:
Is there a reason this uses `log2_size_float` instead of `log2_size_addr`?

Reply:
Yeah I guess this might be clearer with log2_size_addr.